walk/config.go

```diff
@@ -48,34 +49,54 @@ func (wc *walkConfig) isExcluded(rel, base string) bool {
 		return true
 	}
 	f := path.Join(rel, base)
-	for _, x := range wc.excludes {
-		matched, err := doublestar.Match(x, f)
-		if err != nil {
-			// doublestar.Match returns only one possible error, and only if the
-			// pattern is not valid. During the configuration of the walker (see
-			// Configure below), we discard any invalid pattern and thus an error
-			// here should not be possible.
-			log.Panicf("error during doublestar.Match. This should not happen, please file an issue https://github.com/bazelbuild/bazel-gazelle/issues/new: %s", err)
+	isIncluded := false
+	if len(wc.includes) == 0 {
+		isIncluded = true
+	} else {
+		for _, x := range wc.includes {
+			matched := must(doublestar.Match(x, f))
+			if matched {
+				isIncluded = true
+				break
+			}
 		}
+	}
+	if !isIncluded {
+		return true
+	}
+	for _, x := range wc.excludes {
+		matched := must(doublestar.Match(x, f))
```

Comment:
Nit: overall these additional if-else clauses make the func harder to read.

I would recommend try refactoring this into `if {... <return early>}` instead.